kratos/processes/from_json_check_result_process.cpp

```diff
@@ -0,0 +1,866 @@
+        } else {
+            KRATOS_WARNING("FromJSONCheckResultProcess") << "Error checking for variable " << rVariableName << " Component " << ComponentIndex << ", results:\n" << rEntityType << " " << EntityId << " " << std::setprecision(mRelevantDigits) << ValueEntity << "!=" << std::setprecision(mRelevantDigits) << ValueJSON << "; rel_tol=" << mRelativeTolerance << ", abs_tol=" << mAbsoluteTolerance << std::endl;
+        }
+    } else {
+        if (GPIndex > -1) {
+            KRATOS_WARNING("FromJSONCheckResultProcess") << "Error checking for variable " << rVariableName << " in GP " << GPIndex << ", results:\n" << rEntityType << " " << EntityId << " " << std::setprecision(mRelevantDigits) << ValueEntity << "!=" << std::setprecision(mRelevantDigits) << ValueJSON << "; rel_tol=" << mRelativeTolerance << ", abs_tol=" << mAbsoluteTolerance << std::endl;
+        } else {
+            KRATOS_WARNING("FromJSONCheckResultProcess") << "Error checking for variable " << rVariableName << ", results:\n" << rEntityType << " " << EntityId << " " << std::setprecision(mRelevantDigits) << ValueEntity << "!=" << std::setprecision(mRelevantDigits) << ValueJSON << "; rel_tol=" << mRelativeTolerance << ", abs_tol=" << mAbsoluteTolerance << std::endl;
+        }
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+std::string FromJSONCheckResultProcess::GetNodeIdentifier(NodeType& rNode)
+{
+    if (this->Is(USE_NODE_COORDINATES)) {
+        const SizeType digits = 6;
+        std::stringstream ss;
+        ss.setf( std::ios::fixed );
+        ss << "X_" << std::setprecision( digits ) << rNode.X0() << "_Y_" << std::setprecision( digits ) << rNode.Y0() << "_Z_" << std::setprecision( digits ) << rNode.Z0();
+        return ss.str();
+    } else {
+        return std::to_string(rNode.Id());
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+FromJSONCheckResultProcess::NodesArrayType& FromJSONCheckResultProcess::GetNodes(const Flags* pFlag)
+{
+    KRATOS_TRY;
+
+    if (this->IsNot(NODES_CONTAINER_INITIALIZED)) {
+        auto& r_nodes_array = this->Is(CHECK_ONLY_LOCAL_ENTITIES) ? mrModelPart.GetCommunicator().LocalMesh().Nodes() : mrModelPart.Nodes();
+        const auto it_node_begin = r_nodes_array.begin();
+        const int number_of_nodes = static_cast<int>(r_nodes_array.size());
+
+        #pragma omp parallel firstprivate(number_of_nodes)
```

Comment:
```suggestion
        #pragma omp parallel
```
I think the `firstprivate` clause is not needed here.

Reply:
It is not, but @RiccardoRossi recommends to use it

Reply:
@RiccardoRossi can you explain?